File: C/arrary.c
#include <stdio.h>
// An arraray is a collection of items stored in contiguous memory locations.
// Elements in can be accessed by their index.

int main(void) {
    int my_array[5]; // Declare an array of 5 integers
    int my_array_b[] = {1, 2, 3, 4, 5}; // Declare and initialize an array of 5 integers

    size_t length = sizeof(my_array) / sizeof(my_array[0]); // Get the length of the array
    for(size_t i = 0; i < length; i++) {
        my_array[i] = i; // Assign values to the array
    }
    printf("%d\n", my_array[2]); // Print the value at index 2
    return 0;
}

File: Go/arrary.go
package main

import "fmt"

// Basic example of an arrary in Go
func main() {
	// Declare an array of integers
	var numbers [5]int

	// Initialize the array elements
	numbers[0] = 10
	numbers[1] = 20
	numbers[2] = 30
	numbers[3] = 40
	numbers[4] = 50

	// Print the array elements
	fmt.Println(numbers)

	// Accessing array elements
	fmt.Println(numbers[0]) // Output: 10
	fmt.Println(numbers[4]) // Output: 50

	// Modifying array elements
	numbers[1] = 25
	fmt.Println(numbers[1]) // Output: 25

	// Length of an array
	fmt.Println(len(numbers)) // Output: 5
}
